libtelemetry/src/main/java/com/mapbox/android/telemetry/TelemetryService.java

```diff
@@ -239,6 +239,10 @@ private void unregisterTelemetryReceiver() {
 
   private void disableTelemetryLocationState() {
     if (isLocationEnablerFromPreferences) {
+      if (telemetryLocationEnabler == null) {
+        telemetryLocationEnabler = new TelemetryLocationEnabler(true);
```

Comment:
Could you extract a _Lazy initialization_ method as suggested here https://github.com/mapbox/mapbox-events-android/issues/147#issuecomment-392010793?
> We could extract a _Lazy initialization_ method and use it accordingly in both call sites (when enabling and disabling).